src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts

```diff
@@ -217,6 +217,17 @@ export class OfficeCatADI2Page extends OfficeBasePageComponent implements OnInit
     super.ionViewDidLeave();
   }
 
+  displayFuelEfficient(): boolean {
+    let display: boolean;
+    combineLatest([
+      this.pageState.seriousFaults$,
+      this.pageState.dangerousFaults$,
+      this.pageState.adi2DrivingFaults$]).subscribe((data) => {
+      display = data[0].length === 0 && data[1].length === 0 && data[2].length === 0;
+    });
+    return display;
+  }
+
```

Comment:
Lets try and move this to the pageState object and reference this in the HTML using async pipe.

Reply:
i.e. Inside page state, create a new property called 

```
  displayFuelEfficient$: Observable<boolean>;
```

We could try something like this
```
      displayFuelEfficient$: currentTest$.pipe(
        withLatestFrom(
          this.pageState.seriousFaults$,
          this.pageState.dangerousFaults$,
          this.pageState.adi2DrivingFaults$,
        ),
        map((
          [, seriousF, dangerousF, drivingF],
        ) => !!(seriousF?.length === 0 && dangerousF?.length === 0 && drivingF?.length === 0)),
      ),
```

And then in the template

```
pageState.displayFuelEfficient | async
```